docs/development/js-coding-style.md

```diff
@@ -0,0 +1,148 @@
+`HTMLDivElement` over `HTMLElement` and so on.
+
+#### 3. ESLint Configuration Comments
+
+As will be described in a following section, we are using ESLint to automatically check and/or apply our code-style.
+This tool allows developers to disable or alter the linting process through so-called
+[configuration-comments](https://eslint.org/docs/latest/use/configure/rules#using-configuration-comments-1). These
+comments must not be used, since this would ignore code-style for particular sections altogether.
+
+#### 4. Naming Conventions
+
+The naming conventions of the Airbnb code-style only go so far, which is why in ILIAS these rules are extended by the
+following additions:
+
+- Object properties: Airbnb allows space to name properties in snake_case, which would allow snake_case `class` methods
+  because classes are simply objects. Therefore, we require ALL object properties to be named in camelCase.
+- Configurations: ILIAS is using various 3rd-party libraries, which often require configuration-files. To quickly
+  identify these files, we require them to be named in this fashion: `<library>.config.js`
+- Minification: To distinguish between normal JavaScript files and minified versions, we require them to be named like
+  their input-file, but ending with `.min.js`.
+- Testing: Since JavaScript unit tests will not export anything, we require them to be named like the file they are
+  testing.
+
+### Exceptions
+
+Like there are some rules which are not clear enough or incomplete, there are also some rules which are not quite
+applicable in the ILIAS project. Therefore, the following rules are exempt from our code-style:
+
+#### 1. Exempt JavaScript files
+
+There are some JavaScript files which are auto-generated, which cannot comply with the code-style rules. Such files are
+created by e.g. module-bundling or minification. Therefore files matching the following criteria are exempt from all
+code-style rules EXCEPT for addition 1, the copyright notice.
+
+- files located in a `dist/` folder, which as described is the default output folder for bundled files as described in
+  the [bundling guide](./js-bundling.md).
+- files ending with `.min.js`, which are minified versions of the original file.
+
+#### 2. Function Expressions
+
+Airbnb requires ALL function expressions to be named, which is not always required. In ILIAS we therefore weakened this
+rule to only require named function expressions, if they are assigned to a variable. This leaves room for anonymous
+functions (which are not arrow-functions), which are often used as callbacks for event-handlers or IIFE's.
+
+#### 3. Function Declarations
+
+Airbnb requires ALL functions to be declared BEFORE they are used, which is unnecessary due to the fact JavaScript is
+compiled before it is executed and functions are [hoisted](https://developer.mozilla.org/en-US/docs/Glossary/Hoisting),
```

Comment:
I’d suggest to remove the “compiled before it is executed“ part, because AFAIK this is implementation-specific. Someone else probably knows better than me and can comment on this…

Reply:
@nicoroeser the exception or the explanation?

Reply:
What I mean is that JavaScript is not necessarily _compiled_. The notice about hoisting is relevant.